src/development/docker-compose.yml

```diff
@@ -35,6 +35,18 @@ services:
       context: ./../Altinn.Platform/Altinn.Platform.PDF/
     extra_hosts:
       - "host.docker.internal:host-gateway"
+  altinn_pdf_service:
+    container_name: altinn-pdf-service
+    image: browserless/chrome:1-puppeteer-19.2.2
+    restart: always
+    profiles:
+        - pdf
+    networks:
+      - altinntestlocal_network
+    ports:
+      - "3000:3000"
+    extra_hosts:
+      - "${TEST_DOMAIN:-local.altinn.cloud}:host-gateway"
```

Comment:
Why is this `extra_hosts:` declaration different than the other containers?

Reply:
when the pdfservice is generating the PDF it executes a request to the application. The URI to the application is by default local.altinn.cloud/... but as the DNS record resolves to 127.0.0.1 it would not be routed correctly inside the pdfservice container. So i add a extra record to the containers /etc/hosts file to make the request work from inside the container

Reply:
Thanks for the explanation! Pretty obvious in retrospect.

How does this work for users that has set `ALTINN3LOCAL_PORT` to something else than `80`? We have an issue with some of the computers at Finanstilsynet where a system service is using port 80 and thus blocking altinn from using that port.

Reply:
That's something we need to take into account when we implement the code in app-lib-dotnet so it respects ports. This will make sure all requests to local.altinn.cloud will be directed to the host-gateway and be routed through nginx (as long as we take the port into account in our backendcode, I will make a note of this in the issue so we don't forget).